RNN.lua

```diff
@@ -450,7 +448,12 @@ function RNN:accGradParameters(input, gradOutput, scale)
    assert(input:size(2) == self.miniBatch, 'input has incorrect minibatch size!')
    assert(input:size(3) == self.inputSize, 'input has incorrect size!')
 
-   assert(gradOutput:isSameSizeAs(self.output), 'gradOutput has incorrect size!')
+    if (self.batchFirst) then
```

Comment:
Don't you want to make lines 451-456 similar to lines 369-370? Otherwise looks good, you are right, x is used in accGradParameters.